lib/tree_sitter/query_matches.rb

```diff
@@ -23,5 +23,17 @@ def each(&_block)
         end
       end
     end
+
+    # Return all the results as and array of hashes of capture name => node.
+    #
+    # @return [Array<Hash<String, TreeSitter::Node>]
+    def collect
```

Comment:
In Ruby, IIRC, `collect` is an alias for `map`.  Is it different here?  In which case highlighting the difference would be helpful.

Reply:
Yes, you're right, never seen it before though https://ruby-doc.org/3.2.2/Enumerable.html#method-i-collect

It's not the same as `map` here thought. The latter returns an array of `TreeSitter::Match`.

 What name would you suggest?

Reply:
There's a typo in the description in the comment.

Instead of making this return an array, shouldn't it be another way to iterate?  Again with `block_given?` of course.

So maybe `each_captures` or something like that?  Or `each_match_captures`.  The plural is annoying, but...

`each_capture_hash`?

Reply:
I went with `each_capture_hash`.